torch/_refs/nn/functional/__init__.py

```diff
@@ -457,6 +459,83 @@ def threshold(
     return torch.where(a <= threshold, value, a)
 
 
+# CompositeImplicitAutograd - don't register decomp
+# No elementwise type promotion here because it breaks a bunch of dtype tests.
+def triplet_margin_loss(
+    anchor: TensorLikeType,
+    positive: TensorLikeType,
+    negative: TensorLikeType,
+    margin: float = 1.0,
+    p: float = 2,
+    eps: float = 1e-6,
+    swap: bool = False,
+    size_average: Optional[bool] = None,
+    reduce: Optional[bool] = None,
+    reduction: str = "mean",
+) -> TensorLikeType:
+    if size_average is not None or reduce is not None:
+        # TODO: raise exception instead of converting value
+        # msg = "size_average and reduce args are deprecated, please use reduction argument."
+        reduction = _get_string_reduction_arg(size_average=size_average, reduce=reduce)
+
+    return torch.nn.functional.triplet_margin_with_distance_loss(
+        anchor=anchor,
+        positive=positive,
+        negative=negative,
+        distance_function=lambda x, y: torch.pairwise_distance(x, y, p, eps),
+        margin=margin,
+        swap=swap,
+        reduction=reduction,
+    )
+
+
+# Pure Python implementation - don't register decomp
+# No elementwise type promotion here because it breaks a bunch of dtype tests.
+def triplet_margin_with_distance_loss(
+    anchor: TensorLikeType,
+    positive: TensorLikeType,
+    negative: TensorLikeType,
+    *,
+    distance_function: Optional[
+        Callable[[TensorLikeType, TensorLikeType], TensorLikeType]
+    ] = None,
+    margin: float = 1.0,
+    swap: bool = False,
+    reduction: str = "mean",
+) -> TensorLikeType:
+    _check_reduction_value(reduction)
+
+    # TODO: Regular PyTorch has this check for triplet_margin_loss but not for
+    # triplet_margin_with_distance_loss.  However, the docs for both suggest
+    # these must have the same shape.
+    a_dim = anchor.dim()
+    p_dim = positive.dim()
+    n_dim = negative.dim()
+    check(
+        a_dim == p_dim and p_dim == n_dim,
+        lambda: f"All inputs should have same dimension but got {a_dim}D, {p_dim}D and {n_dim}D inputs.",
```

Comment:
"All inputs" -> "The anchor, positive, and negative tensors"
"should have same dimension" -> "are expected to have the same number of dimensions"

Also add the parameter names so users know the rank of each tensor, otherwise the message just says that at least one tensor has a distinct rank, but not which one